changelog/pending/20250428--cli-state--add-pulumi-state-reprotect.yaml

```diff
@@ -0,0 +1,4 @@
+changes:
+- type: feat
+  scope: cli/state
+  description: Add pulumi state reprotect
```

Comment:
Yea I think as discussed this should be `protect` -- reprotecting is just one reason you might use it, but in general you might want to protect an arbitrary resource for an arbitrary reason.